cachito/workers/requests.py

```diff
@@ -28,14 +44,15 @@ def get_requests_session(auth=False):
         elif config.cachito_auth_type == "cert":
             session.cert = config.cachito_auth_cert
 
-    retry = Retry(
-        total=5, read=5, connect=5, backoff_factor=1.3, status_forcelist=(500, 502, 503, 504)
-    )
-    adapter = requests.adapters.HTTPAdapter(max_retries=retry)
+    retry_options = {**DEFAULT_RETRY_OPTIONS, **retry_options}
+    adapter = requests.adapters.HTTPAdapter(max_retries=Retry(**retry_options))
     session.mount("http://", adapter)
     session.mount("https://", adapter)
     return session
 
 
-requests_auth_session = get_requests_session(auth=True)
-requests_session = get_requests_session()
+# These sessions are only for connecting to the internal Cachito API
+requests_auth_session = get_requests_session(
```

Comment:
Just a small nitpick: I saw that we have these two sessions being defined here, and we also have the `pkg_requests_session` and the `nexus_requests_session` being defined in different files.

I saw that both `pkg_requests_session` and the `nexus_requests_session` are identical now (they retry only safe methods), and that although the `requests_session` is defined to retry all methods, it is only being used on a single GET method.

So, I was thinking we could have them all on only two sessions defined on this file:

`requests_auth_session` - is authenticated and allows the retry of all methods
`requests_session` - allows the retry of safe methods (replaces `nexus_requests_session` and `pkg_requests_session` )

Anyway, it's just a suggestion, let me know what you think.

Reply:
I would distinguish sessions not based on how they are configured currently, but take to account a potential changes of each case.

It is better to instantiate session as close to a corresponding use case as possible, to not broke open-closed principle and keep possibility to tune everything.

Reply:
I didn't want to go further with that refactoring, because here are the following dependencies:

- `fetch_npm_source depends` calls `set_request_state` and `update_request_with_config_files`
- both `set_request_state` and `update_request_with_config_files` use `requests_auth_session`
- so there is one case for the auth session, which involves different modules
   * `set_request_state` is part of `cachito.workers.tasks.utils`
   * `update_request_with_config_files` is part of `cachito.workers.pkg_managers.general`
   * common package is `cachito.workers`
So until we bring some other mechanism for session instantiation, we need to keep the session instance somewhere in common modules (the `workers.requests` is proper one from that perspective).

Only one thing I can change here it is the name of that session. It could be called `cachito_requests_auth_session` or just `cachito_auth_session`.

Reply:
@brunoapimentel 

So, do I need to change something in the PR or we can postpone that?

Reply:
We can definitely postpone it, I saw it as a minor nitpick, just wanted to leave the suggestion here.